operator-wg/whitepaper-community-review/README.md

```diff
@@ -0,0 +1,875 @@
+
+### Target Audience / Minimum Level of Experience 
+This document is intended for application developers, Kubernetes cluster operators and service providers (internal or external) - who are looking to learn about operators and the problems they can solve. It can also help teams already looking at operators to learn when and where to use them to best effect. It presumes basic Kubernetes knowledge such as familiarity with Pods and Deployments.
+
+## Foundation
+Kubernetes and other orchestrators' success was due to their focus on containers' main capabilities.
+While companies began their journey to cloud native, working with more specific use cases (microservices, stateless applications) made more sense.
+As Kubernetes and other container orchestrators grew their reputation and extensibility, requirements became more ambitious.
+The desire to use orchestrators' full lifecycle capabilities was also transferred to highly distributed data stores.
+
+But Kubernetes primitives were not built to manage state by default.
+Thus by relying on Kubernetes primitives alone, it's difficult to manage stateful application requirements such as replication, failover automation, backup/restore, upgrades, etc which can occur based on events that are too specific.
+
+We can solve the problem of managing state with the Operator Pattern.
+By leveraging Kubernetes built in capabilities such as self-healing and reconciliation and extending those along with application-specific complexities, it is possible to automate any application's lifecycle and operations and turn it into a highly capable offering.
+
+When we hear about Operators, we think of Kubernetes.
+But the idea of an application whose management is entirely automated can be exported to other platforms.
+So with this paper, we aim to bring this concept to a higher level than Kubernetes itself.
+
+### Operator Design Pattern
+This section describes the pattern with abstract concepts, the next section “Kubernetes Operator Definition” will describe the implementations of the pattern in terms of Kubernetes objects and concepts
+
+The operator design pattern defines how to manage application and infrastructure resources using declarative domain-specific knowledge. The goal of the pattern is to reduce the amount of manual imperative work which is required to keep an application in a healthy and well-maintained state.
+
+By using the operator pattern, the knowledge on how to adjust and maintain a resource is captured in code and usually in a single service (also called a controller).
+
+When using an operator design pattern the user should only be required to describe the desired state of the application and resources. A software pre-written should make the necessary changes in the world so it will be in the desired state. The software will also monitor the real state continuously and take actions to keep it healthy and n the same state (preventing drifts)
```

Comment:
```suggestion
When using an operator design pattern the user should only be required to describe the desired state of the application and resources. The operator implementation should make the necessary changes in the world so it will be in the desired state. The operator will also monitor the real state continuously and take actions to keep it healthy and in the same state (preventing drifts).
````